sdk/core/azure-core-serializer-json-gson/src/test/java/com/azure/core/serializer/json/gson/GsonPropertyNameTests.java

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.json.gson;
+
+import com.azure.core.experimental.serializer.PropertyNameSerializer;
+import com.google.gson.GsonBuilder;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import reactor.test.StepVerifier;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Member;
+import java.lang.reflect.Method;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class GsonPropertyNameTests {
```

Comment:
Would love to see some integration tests to go along with these unit tests. Cases where we pass a class into the `PropertyNameSerializer` and also serialize the same class into a `JsonElement` where we then validate that the `JsonElement` contains JSON attributes with the names returned by `PropertyNameSerializer`.